internal/plan/plan.go

```diff
@@ -211,6 +211,59 @@ func (s *Service) Equal(other *Service) bool {
 	return reflect.DeepEqual(s, other)
 }
 
+// ParseCommand returns a service command as two stream of strings.
+// The base command is returned as a stream and the default arguments
+// in [ ... ] group is returned as another stream.
+func (s *Service) ParseCommand() (base, extra []string, err error) {
+	args, err := shlex.Split(s.Command)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	var inBrackets, gotBrackets bool
+
+	for idx, arg := range args {
+		if inBrackets {
+			if arg == "[" {
+				return nil, nil, fmt.Errorf("cannot nest [ ... ] groups")
```

Comment:
The errors in this function lack context. That is, they need to say "cannot parse service %q command: ..." as a prefix so that when the error pops up in the user interface it is understandable. Typically we do that with a defer at the top that works on the global err variable so that we don't need to repeat that every single time.

Reply:
Added in the new commit https://github.com/canonical/pebble/pull/191/commits/d686b6f3a76b21b1cde43b4565dc1fcb565aafcc.